src/avrdude.conf.in

```diff
@@ -1508,6 +1508,18 @@ programmer
   # see below "nibobee"
 ;
 
+programmer
+  id              = "ch341a";
+  desc            = "ch341a programmer";
+  type            = "ch341a";
+  prog_modes      = PM_ISP;
+  connection_type = usb;
+  usbvid          = 0x1A86;
+  usbpid          = 0x5512;
+  usbvendor       = "wch.cn";
+  usbproduct      = "CH341A";
+;
+
```

Comment:
- I miss a URL that tells me what ch341a refers to. That URL could possibly be to a data sheet, or better still to a product or a circuit that explains how an AVR part is supposed to be rigged up with the CH431A chip. The source code contains comments about that, but they are not great either (see other review comments). I would expect some help here in the avrdude.conf in the form of comments. We use the signal names SCK, SDO, SDI and RESET in this file with other programmers, so at the very minimum comment in the programmer how to up these signals with a CH341A and whether or not you need pullups/pulldowns etc.

- There is a shortcoming of this programmer in that it is too fast: it only can program AVR parts with a minimum F_CPU of 5.6 MHz; this rules out all factory-set AVR parts; please mention this shortcoming here

- Ensure that you use the output of `avrdude -cch341a/s` for the PR; this is the canonical form

- The placement of programmers in `avrdude.conf` is not prescribed in general, *but* do **not** put it after a programmer that has a comment that refers to the next programmer (the programmer above says `see below "nibobee"`, and the PR puts the new programmer between that comment and `nibobee`) 

Example (to be completed):
```
#------------------------------------------------------------
# ch341a
#------------------------------------------------------------

# This allows programming with the CH431A chip [url to data sheet]
# The setup should be like this [url]
#
# In summary, connect AVR part signals to CH341A's pins as follows:
# reset   = [xxx]
#   sck   = [xxx]
#   sdo   = [xxx]
#   sdi   = [xxx]
#
# Note: using this setup limits the programmable AVR parts to those with a F_CPU of at least 5.6 MHz

programmer
    id                     = "ch341a";
    desc                   = "ch341a programmer (target AVR must have minimum F_CPU of 5.6 MHz)";
    type                   = "ch341a";
    prog_modes             = PM_ISP;
    connection_type        = usb;
    usbvid                 = 0x1a86;
    usbpid                 = 0x5512;
    usbvendor              = "wch.cn";
    usbproduct             = "CH341A";
;
```

Reply:
The supported scheme is like the following.
CH341A -- AVR ISP Connector
CLK -- SCK
CS -- RST
MOSI (SDO) -- MOSI (SDO)
MISO (SDI) -- MISO (SDI)
GND -- GND
5V -- Vcc

Note: MOSI is called SDO in avrdude; MISO is called SDI.

Reply:
There are no very good URL for the programmer as there are different configurations. The best website I have found is the following.
https://www.onetransistor.eu/2017/08/ch341a-mini-programmer-schematic.html